nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,866 @@
+        The dependent variable (age, sex, IQ, yes/no, etc.).
+        Target variable to predict. Must have exactly as many elements as
+        3D images in niimg.
+
+    param_grid: dict of str to sequence, or sequence of such. Default None
+        The parameter grid to explore, as a dictionary mapping estimator
+        parameters to sequences of allowed values.
+
+        An empty dict signifies default parameters.
+
+        A sequence of dicts signifies a sequence of grids to search, and is
+        useful to avoid exploring parameter combinations that make no sense
+        or have no effect. See scikit-learn documentation for more information.
+
+    Returns
+    -------
+    param_grid: dict of str to sequence, or sequence of such. Sensible default
+    dict has size 1.
+
+    """
+    if param_grid is None:
+        param_grid = {}
+        # define loss function
+        if isinstance(estimator, LogisticRegression):
+            loss = 'log'
+        elif isinstance(estimator, (LinearSVC, _BaseRidge, SVR)):
+            loss = 'squared_hinge'
+        else:
+            raise ValueError(
+                "Invalid estimator. The supported estimators are: {}".format(
+                             list(SUPPORTED_ESTIMATORS.keys()))
+            )
+        # define sensible default for different types of estimators
+        if hasattr(estimator, 'penalty') and (estimator.penalty == 'l1'):
+            min_c = l1_min_c(X, y, loss=loss)
+        else:
+            min_c = 0.5
+
+        if not isinstance(estimator, _BaseRidge):
+            param_grid['C'] = np.array([2, 20, 200]) * min_c
+        else:
+            param_grid['alpha'] = 1. / np.array([1, 10, 100])
+
+    return param_grid
+
+
+def _parallel_fit(estimator, X, y, train, test, param_grid, is_classif, scorer,
+                  mask_img, class_index, screening_percentile=None):
+    """Find the best estimator for a fold within a job.
+    This function tries several parameters for the estimator for the train and
+    test fold provided and save the ones that performs best. These models are
+    used afterwards to build the averaged model.
+
+    """
+    n_features = X.shape[1]
+    # Checking if the size of the mask image allows us to perform feature
+    # screening
+    selector = check_feature_screening(screening_percentile, mask_img,
+                                       is_classif)
+    do_screening = (n_features > 100) and (screening_percentile < 100.)
+
+    X_train = X[train].copy()
+    y_train = y[train].copy()
+    X_test = X[test].copy()
+    y_test = y[test].copy()
+
+    if (selector is not None) and do_screening:
+        X_train = selector.fit_transform(X_train, y_train)
+        X_test = selector.transform(X_test)
+
+    # If there is no parameter grid, then we use a suitable grid (by default)
+    param_grid = _check_param_grid(estimator, X_train, y_train, param_grid)
+    best_score = None
+    for param in ParameterGrid(param_grid):
```

Comment:
I guess it is done manually to store the fitted models and be able to compute the standard deviation of coefficients accross folds. still it might be faster and simpler to use efficient built-in cross-validation of estimators such as `LogisticRegressionCV` and `RidgeCV` and bootstrap only the selected model. moreover, you probably need less folds for setting hyperparameters than you need resamplings for nonparametric estimation of the coefficients' variance